apps/communications/dialer/test/marionette/lib/views/activities/views.js

```diff
@@ -10,15 +10,24 @@ function ActivitiesView(client) {
 
 ActivitiesView.prototype = {
   pickCreateNew: function() {
-      this.client.waitFor(function() {
-        return this.accessors.addNewContactMenuItem.enabled();
-      }.bind(this));
-      var button = this.accessors.addNewContactMenuItem;
-      button.tap();
-
+      this._openElement(this.accessors.addNewContactMenuItem);
+      
       var contacts = this.client.loader.getAppClass('contacts');
       contacts.switchToCreateNewContactActivity();
       return contacts;
+  },
+
+  pickAddToContact: function(){
+      this._openElement(this.accessors.addToContactMenuItem);
+
+      var contacts = this.client.loader.getAppClass('contacts');
+      contacts.switchToApp();
+      return contacts;
+  },
+
+  _openElement: function(accessorToTap){
+      var button = accessorToTap;
+      button.tap();
```

Comment:
Oh, my apologies, I didn't see `contacts` called 2 different functions. I thought both of then did `contacts.switchToApp()`.

You had a good call, that swich() case was too fishy. That was the sign of a bad design. 

In this case, we can live the code as it's present in the PR.